javalib/src/main/scala/java/lang/HexStringParser.scala

```diff
@@ -0,0 +1,228 @@
+package java.lang
+
+import java.util.regex.Pattern
```

Comment:
Unfortunately we don't have `java.util.regex` and therefore can't use regular expressions here. All regular expressions need to be rewritten into handwritten parsers or recognisers.